app/routes.py

```diff
@@ -1 +1,278 @@
+
+TABLE OF CONTENTS:
+    [0] Imports
+    [1] Helper Functions
+    [2] Blueprints
+    [3] *TASK* endpoints
+    [4] *GOAL* endpoints
+'''
+
+
+
+##### IMPORTS #################################################################
+
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+import datetime
+import requests
+import os
+
+
+
+
+##### HELPER FUNCTIONS ########################################################
+
+def validate_object(object_id, object_type):
+    '''
+    Validates the goal or task based on ID and fetches the object from the database.
+        *object_id:  id of a task or goal
+        *object_type: "goal" or "task" depending on endpoint
+        OUTPUT: goal or task object fetched from database.
+    '''
+    try:
+        object_id = int(object_id)
+    except:
+        abort(make_response({"message":f"Object {object_id} invalid"}, 400))
+
+    if object_type == "task":
+        goal_or_task = Task.query.get(object_id)
+    elif object_type == "goal":
+        goal_or_task = Goal.query.get(object_id)
+    if not goal_or_task:
+        abort(make_response({"message":f"Object {object_id} not found"}, 404))
+
+    return goal_or_task
+
+def post_to_slack(title):
+    '''
+    This function posts a completion method to slack with a slackbot when the
+    patch command "complete_task" is run.
+        *title: title of the task or goal (e.g. task.title)
+        OUTPUT: None (run in place)
+    '''
+    PATH = "https://api.slack.com/api/chat.postMessage"
+    SLACK_AUTH_TOKEN = os.environ.get("SLACK_TOKEN")
+
+    query_params = {
+        "channel":"task-list",
+        "text":f'Someone just completed the task {title}'
+    }
+    header = {"Authorization":SLACK_AUTH_TOKEN}
+    requests.post(PATH, params=query_params, headers=header)
+
+def ordered_tasks_query(sort_method):
+    '''
+    Determines the order_by type (if any) for GET all tasks when called.
+        *sort_method: Evaluates to "asc" or "desc".
+            ALWAYS pass in request.args.get("sort") when calling
+        OUTPUT: returns the ordered tasks.
+    '''
+    if sort_method == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif sort_method == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    return tasks
+
+def ordered_goals_query(sort_method):
+    '''
+    Determines the order_by type (if any) for GET all goals when called.
+        *sort_method: Evaluates to "asc" or "desc".
+            ALWAYS pass in request.args.get("sort") when calling
+        OUTPUT: returns the ordered tasks.
+    '''
+    if sort_method == "asc":
+        goals = Goal.query.order_by(Goal.title.asc())
+    elif sort_method == "desc":
+        goals = Goal.query.order_by(Goal.title.desc())
+    else:
+        goals = Goal.query.all()
+    return goals
```

Comment:
Again, you can combine these using the same technique:

```suggestion
def ordered_query(sort_method, object_type):
    '''
    Determines the order_by type (if any) for GET all objects when called.
        *sort_method: Evaluates to "asc" or "desc".
            ALWAYS pass in request.args.get("sort") when calling
        OUTPUT: returns the ordered tasks.
    '''
    if sort_method == "asc":
        model_objs = object_type.query.order_by(object_type.title.asc())
    elif sort_method == "desc":
        model_objs = object_type.query.order_by(object_type.title.desc())
    else:
        model_objs = object_type.query.all()
    return model_objs